lib/srv/reexec.go

```diff
@@ -612,9 +612,96 @@ func (o *osWrapper) startNewParker(ctx context.Context, credential *syscall.Cred
 	return nil
 }
 
-// RunForward reads in the command to run from the parent process (over a
+type forwardHandlerParams struct {
+	addr           string
+	fileDescriptor *os.File
+}
+
+type forwardHandler func(ctx context.Context, params forwardHandlerParams) error
+
+func handleLocalPortForward(ctx context.Context, params forwardHandlerParams) error {
+	if params.fileDescriptor == nil {
+		return trace.BadParameter("parent process did not send a socket")
+	}
```

Comment:
In both of our current handlers we have to do this check; should we do this in `runForward` instead, and always pass a valid `*os.File` to the handler functions?